src/app.js

```diff
@@ -300,32 +300,24 @@ class Messenger extends EventEmitter {
       attachments
     } = message;
 
-    if (this.options.emitGreetings && this.greetings.test(text)) {
-      const firstName = session.profile && session.profile.first_name.trim() || '';
-      const surName = session.profile && session.profile.last_name.trim() || '';
-      const fullName = `${firstName} ${surName}`;
-
-      this.emit('text.greeting', new Response(this, {event, senderId, session, firstName, surName, fullName}));
-      return;
-    }
-
-    if (this.help.test(text)) {
-      this.emit('text.help', new Response(this, {event, senderId, session}));
+    if (this.emitOptionalEvents(event, senderId, session, text)) {
       return;
     }
 
     if (quickReply) {
-      debug('message.quickReply payload: "%s"', quickReply.payload);
+      const text = quickReply.payload;
+      const normalizedText = text.toLowerCase().trim();
+      debug('message.quickReply payload: "%s"', text);
 
-      this.emit('text', new Response(this, {event, senderId, session, source: 'quickReply', text: quickReply.payload}));
-      this.emit('message.quickReply', new Response(this, {event, senderId, session, payload: quickReply.payload}));
```

Comment:
My understanding was `text and `normalizedText` everywhere; I did not think we were going to diverge for the `message` events.